### PR TITLE
Add invite command

### PR DESCRIPTION
**Pull request by JohnNzj**
why so weird!?!?!??!?!

### CODE REVIEW

**Comment by podocarp** [comment]
Pls see old comments in #103

**Comment by Aulud** [comment]
Apologies for blocking this by the View command, but it's actually good to go.

**Review by Aulud — DISMISSED** [review]
Apologies for blocking this by the View command, but in principle it's good to go.

File: src/main/java/seedu/address/logic/commands/InviteCommand.java [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+            NameContainsAllKeywordsPredicate predicate = new NameContainsAllKeywordsPredicate(keywords);
+
+            findResult = model.findPersonAll(predicate);
+            assert findResult != null : "List of people in contacts should not be null.";
+
+            if (findResult.size() != 1) {
+                String warning = String.format(MESSAGE_NON_UNIQUE_SEARCH_RESULT, name);
+                warningMessage.append(warning).append("\n");
+                continue;
+            }
```
Comment (liakify) [inline src/main/java/seedu/address/logic/commands/InviteCommand.java:98]: At some point we probably should abstract out this name searching logic from activity, invite, expense (and disinvite?) commands cause its currently being repeated a lot.

Also we need to fix the issue of names being impossible to access if they are a sub-string of other names so I guess that can be done together in one go

File: src/test/java/seedu/address/logic/commands/InviteCommandTest.java [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+        InviteCommand inviteCommand = new InviteCommand(peopleToInvite);
+        inviteCommand.execute(model);
+
+        assertEquals(1, activity.getParticipantIds().size());
+
+        peopleToInvite.add(BENSON.getName().fullName);
+        inviteCommand = new InviteCommand(peopleToInvite);
+        inviteCommand.execute(model);
+
+        assertEquals(2, activity.getParticipantIds().size());
```
Comment (liakify) [inline src/test/java/seedu/address/logic/commands/InviteCommandTest.java]: might be less prone to bugs to assert participant IDs lists are equal

File: src/test/java/seedu/address/logic/commands/InviteCommandTest.java [hunk 3]
```diff
@@ -0,0 +1,139 @@
...
+        inviteCommand.execute(model);
+
+        assertEquals(1, activity.getParticipantIds().size());
+
+        peopleToInvite.add(BENSON.getName().toString());
+        peopleToInvite.add(BENSON.getName().toString());
+        inviteCommand = new InviteCommand(peopleToInvite);
+        inviteCommand.execute(model);
+
+        assertEquals(2, activity.getParticipantIds().size());
```
Comment (liakify) [inline src/test/java/seedu/address/logic/commands/InviteCommandTest.java]: same thing, assert equal participant ID list should be safer

File: src/test/java/seedu/address/logic/commands/InviteCommandTest.java [hunk 4]
```diff
@@ -0,0 +1,139 @@
...
+        InviteCommand inviteCommand = new InviteCommand(peopleToInvite);
+        inviteCommand.execute(model);
+
+        assertEquals(1, activity.getParticipantIds().size());
+
+        peopleToInvite.add(BENSON.getName().fullName);
+        inviteCommand = new InviteCommand(peopleToInvite);
+        inviteCommand.execute(model);
+
+        assertEquals(2, activity.getParticipantIds().size());
```
Comment (liakify) [inline src/test/java/seedu/address/logic/commands/InviteCommandTest.java]: and here

File: src/test/java/seedu/address/logic/commands/InviteCommandTest.java [hunk 5]
```diff
@@ -0,0 +1,139 @@
...
+        InviteCommand inviteCommand = new InviteCommand(peopleToInvite);
+        inviteCommand.execute(model);
+
+        assertEquals(0, activity.getParticipantIds().size());
+
+        peopleToInvite.add(BENSON.getName().fullName);
+        inviteCommand = new InviteCommand(peopleToInvite);
+        inviteCommand.execute(model);
+
+        assertEquals(0, activity.getParticipantIds().size());
```
Comment (liakify) [inline src/test/java/seedu/address/logic/commands/InviteCommandTest.java:150]: and here

**Review by liakify — DISMISSED** [review]
should be fine, not sure if you want to change the test

**Review by liakify — APPROVED** [review]
Looks good

**Pull request merged**